### PR TITLE
Add API version apps/v1, and bump DaemonSet to apps/v1

### PR DESCRIPTION
**Pull request by janetkuo**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: kubernetes/features#484

**Special notes for your reviewer**: This PR targets `master`, as a backup if #53223 (targeting features branch) falls through 

@kubernetes/sig-apps-api-reviews 

**Release note**:

```release-note
Add API version apps/v1, and bump DaemonSet to apps/v1
```

### CODE REVIEW

File: pkg/apis/apps/v1/conversion_test.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	}{
+		"RollingUpdateDaemonSet Conversion 2": {
+			rollingUpdateDs1: &extensions.RollingUpdateDaemonSet{MaxUnavailable: intorstr},
+			rollingUpdateDs2: &appsv1.RollingUpdateDaemonSet{MaxUnavailable: &intorstr},
+		},
+	}
+
+	for k, tc := range testcases {
+		// extensions -> v1
+		internal1 := &appsv1.RollingUpdateDaemonSet{}
```
Comment (kow3ns) [inline pkg/apis/apps/v1/conversion_test.go]: Do we want to test conversions between v1beta2 or are we confident that transitivity covers this?

File: pkg/apis/apps/v1/doc.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// +k8s:conversion-gen=k8s.io/kubernetes/pkg/apis/apps
+// +k8s:conversion-gen=k8s.io/kubernetes/pkg/apis/extensions
+// +k8s:conversion-gen-external-types=../../../../vendor/k8s.io/api/apps/v1
+// +k8s:defaulter-gen=TypeMeta
+// +k8s:defaulter-gen-input=../../../../vendor/k8s.io/api/apps/v1
```
Comment (kow3ns) [inline pkg/apis/apps/v1/doc.go]: apps/v1beta2 conversions?

File: staging/src/k8s.io/api/apps/v1/types.go [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+	// out of date by some window of time.
+	// Populated by the system.
+	// Read-only.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#spec-and-status
+	// +optional
+	Status DaemonSetStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+const (
+	// DEPRECATED: DefaultDaemonSetUniqueLabelKey is used instead.
```
Comment (kow3ns) [inline staging/src/k8s.io/api/apps/v1/types.go]: Do we still want to include this in v1?

**Review by kow3ns — APPROVED** [review]

File: pkg/apis/apps/v1/conversion_test.go (hunk 1 above)
Comment (janetkuo) [reply pkg/apis/apps/v1/conversion_test.go]: Conversion only happens between internal and external version, i.e. extensions/ <--> apps/v1, but not apps/v1beta2 <--> apps/v1

File: staging/src/k8s.io/api/apps/v1/types.go (hunk 3 above)
Comment (janetkuo) [reply staging/src/k8s.io/api/apps/v1/types.go]: Oops this can be removed from here and apps/v1beta2

**Comment by smarterclayton** [comment]
No significant API concerns from me.  Have we assessed the relevant open daemonset issues to determine if there are any known gaps from users in terms of clarity of field name that need to be addressed prior to promotion?

File: pkg/apis/apps/v1/doc.go (hunk 2 above)
Comment (janetkuo) [reply pkg/apis/apps/v1/doc.go]: (ditto)

**Comment by kow3ns** [comment]
@smarterclayton we can not find an open issues and have received no feedback that indicates confusion.

File: pkg/apis/apps/v1/conversion_test.go (hunk 1 above)
Comment (kow3ns) [reply pkg/apis/apps/v1/conversion_test.go]: ack

File: api/swagger-spec/apps_v1.json [hunk 4]
```diff
@@ -0,0 +1,3784 @@
...
+      "type": "string",
+      "description": "Kind is a string value representing the REST resource this object represents. Servers may infer this from the endpoint the client submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds"
+     },
+     "apiVersion": {
+      "type": "string",
+      "description": "APIVersion defines the versioned schema of this representation of an object. Servers should convert recognized schemas to the latest internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#resources"
+     },
+     "metadata": {
+      "$ref": "v1.ListMeta",
+      "description": "Standard list metadata.
```
Comment (dixudx) [inline api/swagger-spec/apps_v1.json]: @janetkuo Please rebase the master branch and regenerate this file to get it updated. Or simple `sed "s/mulitple/multiple/"`

@saad-ali FYI. This is causing batch `pull-kubernetes-verify` failed in issue #53481.

**Comment by janetkuo** [comment]
Rebased. Re-applying tags

**Comment by janetkuo** [comment]
Another rebase :/

**Pull request merged**